app/code/core/Mage/Core/Block/Messages.php

```diff
@@ -247,6 +247,7 @@ public function getGroupedHtml()
         if ($html) {
             $html .= '</' . $this->_messagesFirstLevelTagName . '>';
         }
+        $this->_messages = null;
```

Comment:
```suggestion
        $this->getMessageCollection()->clear();
```

Reply:
Yes, I don't remember if I tested or not this solution before.
Working for now, I continue testing it.